crypto/ml_dsa/ml_dsa_sign.c

```diff
@@ -111,12 +113,12 @@ static int ml_dsa_sign_internal(const ML_DSA_KEY *priv, int msg_is_mu,
     if (!matrix_expand_A(md_ctx, priv->shake128_md, priv->rho, &a_ntt))
         goto err;
     if (msg_is_mu) {
-        if (encoded_msg_len != mu_len)
+        if (msg_len != mu_len)
             goto err;
-        mu_ptr = (uint8_t *)encoded_msg;
+        mu_ptr = (uint8_t *)msg;
     } else {
-        if (!shake_xof_2(md_ctx, priv->shake256_md, priv->tr, sizeof(priv->tr),
-                         encoded_msg, encoded_msg_len, mu_ptr, mu_len))
+        if (!shake_xof_3(md_ctx, priv->shake256_md, priv->tr, sizeof(priv->tr),
+                         context_msg, context_len, msg, msg_len, mu_ptr, mu_len))
```

Comment:
Note here that this will do a update with a count of 0 zero when context_len is zero.

Reply:
That should be a noop afaik ?

Reply:
I checked for good measure, it is a noop as I expected, so this works as planned